craft_providers/bases/buildd.py

```diff
@@ -301,6 +307,41 @@ def _ensure_instance_config_compatible(
             "Instance is compatible with compatibility tag %r", config.compatibility_tag
         )
 
+    def _ensure_setup_completed(
+        self, *, executor: Executor, deadline: Optional[float]
+    ) -> None:
+        """Ensure the instance was fully setup.
+
+        The last step of setting up an instance is to set the `setup` key in the
+        instance config file to True. This flag is used to verify that setup was
+        completed for the instance.
+
+        :raises BaseCompatibilityError: If setup was not completed.
+        """
+        _check_deadline(deadline)
+
+        try:
+            config = InstanceConfiguration.load(
+                executor=executor,
+                config_path=self.instance_config_path,
+            )
+        except ValidationError as error:
+            raise BaseCompatibilityError(
+                reason="failed to parse instance configuration file",
+            ) from error
+        except FileNotFoundError as error:
+            raise BaseCompatibilityError(
+                reason="failed to find instance config file",
+            ) from error
+
+        if config is None:
+            raise BaseCompatibilityError(reason="instance config is empty")
+
+        if not config.setup or config.setup is False:
```

Comment:
So in the end what was the decision regarding auto-cleaning incomplete instances?